vertx-sql-client/src/main/java/io/vertx/sqlclient/spi/Driver.java

```diff
@@ -52,5 +53,32 @@
    * @return true if the driver accepts the {@code connectOptions}, false otherwise
    */
   boolean acceptsOptions(SqlConnectOptions connectOptions);
-
+  
+  /**
+   * @param connectionUri a connection URI
+   * @return true if the driver accepts the {@code uri}, false otherwise
+   */
+  boolean acceptsUri(String connectionUri);
+  
+  /**
+   * Returns a specific {@link SqlConnectOptions} instance corresponding to the {@code uri}.
+   * 
+   * @param <T> the expected type of {@code SqlConnectOptions}
+   * @param connectionUri a connection URI
+   * @return connect options specific to the driver
+   * @throws IllegalArgumentException if the driver does not accept the {@code uri}
+   */
+  <T extends SqlConnectOptions> T getConnectOptions(String connectionUri);
+  
+  /**
+   * Returns a specific {@link SqlConnectOptions} instance corresponding to the {@code uri} and 
+   * initialized with {@code json}.
+   * 
+   * @param <T> the expected type of {@code SqlConnectOptions}
+   * @param connectionUri a connection URI
+   * @param json a json
+   * @return connect options specific to the driver
+   * @throws IllegalArgumentException if the driver does not accept the {@code uri}
+   */
+  <T extends SqlConnectOptions> T getConnectOptions(String connectionUri, JsonObject json);
```

Comment:
we should try to remove this as it always do the same thing

Reply:
Ok now that that I got back into it, I get it. Yes indeed `connectionUri` and `JsonObject` does more or less the same thing so either we decide to use them together (merge) for convenience as I suggested or we can use one or the other defining two methods. You're probably right it seems more consistent to do that way.